### PR TITLE
Make the project installable as a pytest plugin

### PR DESCRIPTION
**Pull request by ocaballeror**
By submitting a PR to this repository, you agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/auth0/.github/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

> Describe the purpose of this PR along with any background information and the impacts of the proposed change. For the benefit of the community, please do not assume prior context.
>
Main purpose is to make this project installable as a python package and discoverable by pytest's plugin system, instead of asking users to copy the files into their projects.
> Provide details that support your chosen implementation, including: breaking changes, alternatives considered, changes to the API, etc.
>
The only opinionated and "breaking" change is to name the package pytest-aioboto3 instead of aioboto3-testing. This is to follow the standard naming of pytest plugins, and to make it easier to discover by users. The actual test file `test_s3.py` is kept in the tests directory to separate it from the main code.

I also added support for python 3.7+, because the only 3.9 feature that was being used was the typed `Popen`, but that is easily fixed with `from __future__ import annotations`.
> If the UI is being changed, please provide screenshots.

### Testing

> Describe how this can be tested by reviewers. Be specific about anything not tested and reasons why. If this library has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
```shell
poetry install
poetry run pytest
```
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.
>
> Also include details of the environment this PR was developed in (language/platform/browser version).

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `master`

### CODE REVIEW

File: pytest_aioboto3/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+from .aioboto3_fixtures import moto_patch_session, aioboto3_s3_client
+from .moto_fixtures import moto_services
+
+__all__ = ["moto_pathc_session", "aioboto3_s3_client", "moto_services"]
```
Comment (arnaldojvg) [inline pytest_aioboto3/__init__.py:4]: typo
```suggestion
__all__ = ["moto_patch_session", "aioboto3_s3_client", "moto_services"]
```

**Review by arnaldojvg — APPROVED** [review]
Hey, this looks pretty good! Looking forward to have this available on Pypi :)

**Review by phillipuniverse — CHANGES_REQUESTED** [review]
Really awesome stuff! Thanks so much for contributing!!!

> The only opinionated and "breaking" change is to name the package pytest-aioboto3 instead of aioboto3-testing. This is to follow the standard naming of pytest plugins, and to make it easier to discover by users. The actual test file test_s3.py is kept in the tests directory to separate it from the main code.

I agree, great improvement. I went ahead and changed the name of the repo.

You know, I hadn't really planned on putting this on PyPi but I guess I didn't realize how close this was! Thanks for cleaning this up and I will try to get this up on PyPi in the next week or so!

File: pytest_aioboto3/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ocaballeror) [reply pytest_aioboto3/__init__.py:4]: Correct. The pytest11 entrypoint is what makes this a pytest plugin. Pytest will try to import everything from the package that we specified in the entrypoint, so we need to declare the fixtures in the main `__init__.py` file. Only fixtures that are referenced here will be automatically discovered by pytest.

File: pyproject.toml [hunk 2]
```diff
@@ -3,19 +3,20 @@ requires = ["poetry-core>=1.0.0"]
...
 version = "0.1.0"
 description = "Aioboto3 Pytest with Moto"
 authors = ["Shipwell Engineering", "Phillip Verheyden"]
 
 [tool.poetry.dependencies]
-python = "^3.9"
+python = "^3.7"
 aioboto3 = "^9.5.0"
+moto = {extras = ["s3", "server"], version = "^3.1.6"}
+types-aiobotocore = {extras = ["s3"], version = "^2.2.0"}
```
[the remark this answers is not in the document]
Comment (ocaballeror) [reply pyproject.toml:15]: Well, there is an explicit import of `types_aiobotocore_s3` [here](https://github.com/phillipuniverse/pytest-aioboto3/blob/main/tests/aioboto3_fixtures.py#L10), so the package NEEDS to be installed or there will be an import error.

The way I see it, dev dependencies are those that are only needed to run the tests or set up the project locally, while the main dependencies are those required to install and run the main package.

**Review by phillipuniverse — APPROVED** [review]
Great stuff, thanks for contributing!

**Pull request merged**